tests/host_tools/network.py

```diff
@@ -38,9 +38,9 @@ def __init__(self, ssh_config):
 
         self._init_connection()
 
-    def execute_command(self, cmd_string):
+    def execute_command(self, cmd_string, with_asyncio=True):
```

Comment:
What's the difference between executing a command with asyncio and without asyncio ? Why do we need this ?

Reply:
The issue is not that the command ran on the same threads. In practice, `run_cmd` will start iperf3 server processes in the background (on the host), on the main thread, and we do not account for them. We account for the guest processes, started with `execute_command`. Those commands have to be tracked in order to get their `stdout`. This has to happen on different threads, while each thread monitors a specific iperf3 client stdout and returns it to the main thread. Then the issue here is that the inner implementation of `execute_command` depends on asyncio runtime, which provides an incompatible interface to be used within threads, because of [1].

[1] https://python.readthedocs.io/en/latest/library/asyncio-dev.html#concurrency-and-multithreading

Reply:
We should understand what causes this behaviour. Why running guest commands with asyncio can lead to deadlocks. I don't think this is the desired behavior.

Your code works around the problem for this specific scenario. But we should make sure we don't encounter this issue in other scenarios as well.